pkg/apis/v1alpha1/handlers/products.go

```diff
@@ -2,16 +2,49 @@ package handlers
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/caicloud/nirvana-practice/pkg/apis/meta/v1"
 	api "github.com/caicloud/nirvana-practice/pkg/apis/v1alpha1"
 	"github.com/caicloud/nirvana-practice/pkg/errors"
 )
 
-func GetProduct(ctx context.Context, name string) (*api.Product, error) {
+var defaultProduct = &api.Product{
+	Metadata: v1.Metadata{
+		Name: "Apple",
+	},
+	Spec: &api.ProductSpec{
+		Category: "Fruit",
+	},
+}
+
+func CreateProduct(ctx context.Context, product *api.Product) (*api.Product, error) {
 	return nil, errors.ErrorNotImplemented.Error()
 }
 
 func ListProducts(ctx context.Context, options *v1.ListOptions) (*api.ProductsList, error) {
-	return nil, errors.ErrorNotImplemented.Error()
```

Comment:
I see that all incomplete functions have been implemented. Doesn't that defeat the purpose of this repo?

nirvana-practice is a public repo meant for helping people practice writing web server with Nirvana. If we finish everything, then they have nothing to practice...

Reply:
Have removed the implementation.